blueocean-personalization/src/main/js/components/DashboardCards.jsx

```diff
@@ -56,188 +36,44 @@ export class DashboardCards extends Component {
         }
     }
 
-    _onRunAgainClick(pipeline) {
-        this.props.replayPipeline(pipeline);
-
-        const name = decodeURIComponent(pipeline.name);
-
-        toastService.newToast({
-            text: `Queued "${name}"`,
-        });
-    }
-
-    _onRunClick(pipeline) {
-        this.props.runPipeline(pipeline);
-
-        const name = decodeURIComponent(pipeline.name);
-
-        toastService.newToast({
-            text: `Queued "${name}"`,
-        });
-    }
-
-    _onStopClick(pipeline) {
-        this.props.stopPipeline(pipeline);
-
-        const name = decodeURIComponent(pipeline.name);
-        const runId = pipeline.latestRun.id;
-
-        toastService.newToast({
-            text: `Stopping "${name}" #${runId}...`,
-        });
-    }
-
```

Comment:
The Run/ReplayButton components do all the work of run, stop and replay... so this code can go.